### ISSUE TITLE
Playback Stalls For Few Seconds When Playback Initiated

### ISSUE THREAD

**Issue by waxidiotic** [google/shaka-player#1666]
**Have you read the [FAQ](https://bit.ly/ShakaFAQ) and checked for duplicate open issues?**:

Yes

**What version of Shaka Player are you using?**:

* v2.3.5 in the current production version of JW Player
* Also tested with v2.4.5, v2.5.0-beta2 and the current master branch

**Can you reproduce the issue with our latest release version?**:

Yes

**Can you reproduce the issue with the latest code from `master`?**:

Yes

**Are you using the demo app or your own custom app?**:

Reproduced with both

**If custom app, can you reproduce the issue using our demo app?**:

n/a

**What browser and OS are you using?**:

* macOS Mojave 10.14.1
* Chrome 70.0.3538.102
* Firefox 63.0.1
* Safari 12.0.1

**What are the manifest and license server URIs?**:

* Manifest: https://cdn-test.online.meo.pt/sdash/LIVE%24SIC_Noticias/index.mpd/Manifest?device=DASH&start=LIVE&end=END
* No license server necessary

**What did you do?**

* In JW Player, configure the player to autostart
* In Shaka demo app or using just the compiled version, click Play as soon as player appears

**What did you expect to happen?**

* If the pause button appears to denote that the player is playing, I expect the playback to immediately begin

**What actually happened?**

* Video hangs on first frame for upwards of 5 seconds before beginning
* No audio can be heard during these few seconds
* Using the debug compiled version with V2 logging, nothing appeared in the console to explain what was happening

**Comment by fadomire** [comment]
i had a similar issue, maybe it is not the same case but i had the ‘’’startAtSegmentBoundary’’’ option set to true

**Comment by joeyparrish** [comment]
I'm getting HTTP 504 errors from your server.  Each one takes fully 30 seconds.  Could it be that the hang is caused by an overwhelmed server, load-balancer, or CDN?  Here's a screenshot:

![1666 - http 504](https://user-images.githubusercontent.com/6843156/48739210-1f128100-ec08-11e8-9e20-cf1e658022a6.png)

Please try again and use the network panel in your browser to monitor the server's response codes and response times.  If this server issue is resolved and you still have playback problems, please let us know and we can look again.

**Comment by waxidiotic** [comment]
@joeyparrish  I'm getting the same (504) response now, but it wasn't this way when I submitted it. Let me reach out to the customer to see what's going on.

**Comment by joeyparrish** [comment]
Okay.  I'll put it back in the triage queue for whoever is on duty after me.

**Comment by waxidiotic** [comment]
Here's an updated URL: https://cdn-test.online.meo.pt/sdash/LIVE%24Trailers_VideoClube/index.mpd/Manifest?device=DASH&start=LIVE&end=END

**Comment by joeyparrish** [comment]
With the new URL, I can't see anything obviously wrong.  The times below are measured from my very fast workstation on my very fast internet connection, so they may vary and could be much longer for others.

I'm seeing a load latency of about 2-3 seconds as measured by Shaka Player.  You can measure this state yourself by turning setting the log level to "Debug" in the demo app UI under "configuration" and observing the JS console.  Load latency here is the time from the `load()` call to the video element's `loadeddata` event, which corresponds to having anything buffered at all.  This includes fetching and parsing the manifest, fetching and consuming init segments, and fetching and consuming the very first media segments (1 audio, 1 video).  At this point, you should see the first frame of video on the screen and a buffering spinner.

Is this the state which you are calling a stall?

From that point until we exit the buffering state is about 2-3 seconds more for me.  This includes fetching additional media segments to meet the buffering goals expressed in the manifest and the player configuration.  The manifest has a `minBufferTime` attribute of 6 seconds, so we cannot start playback until 6 seconds of media has been buffered.

If it's this initial buffering which takes too long for you, you should reduce the `minBufferTime` attribute in the DASH manifest to something more reasonable.

If you can't control the content, you can override `minBufferTime` with the `manifest.dash.ignoreMinBufferTime` config in the Player.  (This was released in v2.5.0-beta2.  See issue #1547 and PR #1581.)  For example:

```js
// DASH minBufferTime will have no effect if we set this field before loading:
player.configure('manifest.dash.ignoreMinBufferTime', true);
// We can now control initial buffering purely through this field:
player.configure('streaming.rebufferingGoal', 2 /* seconds */);
```

Does this help?

**Comment by waxidiotic** [comment]
@joeyparrish Yeah, that sounds like the point in which it looks like our player is stalling, because we're not displaying the buffer spinner. However, I'm confused why the playback of the stream below starts right away for us in our player:

https://wowzaec2demo.streamlock.net/live/bigbuckbunny/manifest_mpm4sav_mvtime.mpd

It also has a `minBufferTime` of 6 seconds.

**Comment by waxidiotic** [comment]
This is amazing @joeyparrish. Thank you so much. I'll relay this to the customer.

**Comment by joeyparrish** [comment]
Happy to help.  Should we leave the issue open for a few more days?
